### PR TITLE
docs: fix links kubectl docs

### LINKED ISSUE #479: Dead links to pages from kubectl book
The book has been removed: https://github.com/kubernetes/kubectl/commit/7134adff7cdfd2f3ff21c0d31882691969f849b8 and links to its pages are dead now.

For example, check `declarative application management` link on [this page](https://docs.tutor.overhang.io/k8s.html#technical-details).

### PR DESCRIPTION
**Pull request by regisb**
Previous links resulted in 404.

Close #479.

This is ready for review @overhangio/tutor-developers. Cc @0x29a who opened #479.

### CODE REVIEW

File: docs/k8s.rst [hunk 1]
```diff
@@ -57,7 +57,7 @@ Technical details
...
 
-Basically, the whole platform is described in manifest files stored in ``$(tutor config printroot)/env/k8s``. There is also a ``kustomization.yml`` file at the project root for `declarative application management <https://kubectl.docs.kubernetes.io/pages/app_management/apply.html>`_. This allows us to start and update resources with commands similar to ``kubectl apply -k $(tutor config printroot) --selector=...`` (see the ``kubectl apply`` `official documentation <https://kubectl.docs.kubernetes.io/pages/app_management/apply.html>`_).
+Basically, the whole platform is described in manifest files stored in ``$(tutor config printroot)/env/k8s``. There is also a ``kustomization.yml`` file at the project root for `declarative application management <https://kubectl.docs.kubernetes.io/guides/config_management/introduction/#declarative-application-management>`_. This allows us to start and update resources with commands similar to ``kubectl apply -k $(tutor config printroot) --selector=...`` (see the ``kubectl apply`` `official documentation <https://kubectl.docs.kubernetes.io/references/kubectl/apply/>`__).
```
Comment (0x29a) [inline docs/k8s.rst:60]: Is `declarative application management` a one-off hyperlink? If so, double underscore should be used.

**Review by 0x29a — CHANGES_REQUESTED** [review]
@regisb, thanks for fixing this.

I see one more dead link:
```
➜  tutor git:(regisb/fix-kubectl-docs) rg "https://kubectl.docs.kubernetes.io/pages"
docs/k8s.rst
70:To learn more about "kustomizations", refer to the `official documentation <https://kubectl.docs.kubernetes.io/pages/app_customization/introduction.html>`__.
```

**Comment by regisb** [comment]
Thanks for the comments @0x29a! This is ready for review again.

**Review by 0x29a — APPROVED** [review]
Thanks!

**Pull request merged**